src/wp-includes/class-wp-locale-switcher.php

```diff
@@ -196,11 +196,8 @@ private function load_translations( $locale ) {
 		load_default_textdomain( $locale );
 
 		foreach ( $domains as $domain ) {
-			if ( 'default' === $domain ) {
-				continue;
-			}
-
-			unload_textdomain( $domain );
+			// unload_textdomain( $domain ); // TODO: Don't mark as unloaded so it can be reloaded. Is this ok?
+			unset( $l10n[ $domain ] );
```

Comment:
@swissspidy To answer your [question](https://core.trac.wordpress.org/ticket/39210#comment:21), yes it would still be required, although I think we can just call `unset()` here.